presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpiller.java

```diff
@@ -118,7 +118,7 @@ private void writePages(Iterator<Page> pageIterator)
     {
         checkState(writable, "Spilling no longer allowed. The spiller has been made non-writable on first read for subsequent reads to be consistent");
 
-        try (SliceOutput output = new OutputStreamSliceOutput(new FileOutputStream(targetFileName.toFile(), true), BUFFER_SIZE)) {
+        try (SliceOutput output = new OutputStreamSliceOutput(targetFile.newOutputStream(APPEND), BUFFER_SIZE)) {
```

Comment:
Definitely. We append more pages to the spill file. Previous code used append too in `FileOutputStream(.., /* append*/ true)`